### PR TITLE
VSTest task changes

### PR DESCRIPTION
**Pull request by kaadhina**

### CODE REVIEW

File: Tasks/VsTest/VSTest.ps1 [hunk 1]
```diff
@@ -38,9 +43,17 @@ if($testAssemblyFiles)
 {
     Write-Verbose "Calling Invoke-VSTest for all test assemblies"
     $timeline = Start-Timeline -Context $distributedTaskContext
-	$cwd = Get-Location
```
Comment (chrisrpatterson) [inline Tasks/VsTest/VSTest.ps1]: This may not be a good working directory, I am not quite sure what Get-Location is going to return in the context of running this task in the agent.

File: Tasks/VsTest/VSTest.ps1 [hunk 2]
```diff
@@ -38,9 +43,17 @@ if($testAssemblyFiles)
...
+    $buildNumber = Get-Variable -Context $distributedTaskContext -Name "Build.BuildNumber"
+    $buildUri = Get-Variable -Context $distributedTaskContext -Name "Build.BuildUri"	
+    $cwd = $buildDir+"\"+$projectName
+    $testResultsDir = $buildDir+"\"+$projectName+"\"+"TestResults"
+    $trxLocation = $buildDir+"\"+$projectName+"\"+"Temp"
     Write-Verbose "Calling Invoke-VSTest from working folder: $cwd"
-    Invoke-VSTest -TestAssemblies $testAssemblyFiles -Timeline $timeline -VSTestVersion $vsTestVersion -TestFiltercriteria $testFiltercriteria -RunSettingsFile $runSettingsFile -PathtoCustomTestAdapters $pathtoCustomTestAdapters -CodeCoverageEnabled $codeCoverage -OtherConsoleOptions $otherConsoleOptions -WorkingFolder $cwd
+    Invoke-VSTest -TestAssemblies $testAssemblyFiles -Timeline $timeline -VSTestVersion $vsTestVersion -TestFiltercriteria $testFiltercriteria -RunSettingsFile $runSettingsFile -PathtoCustomTestAdapters $pathtoCustomTestAdapters -CodeCoverageEnabled $codeCoverage -OverrideTestrunParameters $overrideTestrunParameters -OtherConsoleOptions $otherConsoleOptions -WorkingFolder $cwd -TestResultsDir $testResultsDir -ResultLocation $trxLocation
```
Comment (chrisrpatterson) [inline Tasks/VsTest/VSTest.ps1]: Are you even populating the timeline anymore?  I don't believe so and if not we should remove it.

File: Tasks/VsTest/task.json [hunk 3]
```diff
@@ -38,30 +38,45 @@
...
+      "name": "configuration", 
+      "type": "string", 
+      "label": "Configuration", 
+      "defaultValue":"", 
+      "required":false 
+    },
+    {
       "name": "runSettingsFile",
-      "type": "filePath",
+      "type": "string",
```
Comment (chrisrpatterson) [inline Tasks/VsTest/task.json]: Why did you change this to a string?  This should be a file path so I optionally get a picker and the infrastructure will automatically try to convert a relative path to a full path.

Comment (bryanmacfarlane) [inline Tasks/VsTest/task.json]: Yes, this should be a filePath

File: Tasks/VsTest/VSTest.ps1 [hunk 4]
```diff
@@ -38,9 +43,17 @@ if($testAssemblyFiles)
 {
     Write-Verbose "Calling Invoke-VSTest for all test assemblies"
     $timeline = Start-Timeline -Context $distributedTaskContext
-	$cwd = Get-Location  
+    $projectName = Get-Variable -Context $distributedTaskContext -Name "System.TeamProject"
+    $buildDir = Get-Variable -Context $distributedTaskContext -Name "Agent.BuildDirectory" -Global $FALSE
+    $buildNumber = Get-Variable -Context $distributedTaskContext -Name "Build.BuildNumber"
+    $buildUri = Get-Variable -Context $distributedTaskContext -Name "Build.BuildUri"	
+    $cwd = $buildDir+"\"+$projectName
+    $testResultsDir = $buildDir+"\"+$projectName+"\"+"TestResults"
```
Comment (chrisrpatterson) [inline Tasks/VsTest/VSTest.ps1]: You don't need project here BuildDir should be unique enough
